src/cxy_ctl.erl

```diff
@@ -653,3 +667,28 @@ get_buffer_times(Buffer_Name, Num_Items) ->
 format_buffer_times({Task_Fun, {_,_,Micro} = Start, Spawn_Elapsed, Exec_Elapsed}) ->
     [{task_fun, Task_Fun}, {start, calendar:now_to_universal_time(Start), Micro},
      {spawn_time_micros, Spawn_Elapsed}, {exec_time_micros, Exec_Elapsed}].
+
+-define(HW_READ_CMD, {?HIGH_WATER_PROCS_POS, 0}).
+-define(HW_RESET_CMD, {?HIGH_WATER_PROCS_POS, -1, 0, 0}).
+
+%% @doc
+%%   Return the highest number of concurrent processes for a given task type.
+%%   The one-argument form just returns the high-water value, while the
+%%   two-argument form allows the value to be reset.
+%% @end
+
+-spec high_water(task_type()) -> non_neg_integer().
+-spec high_water(task_type(), boolean()) -> non_neg_integer().
+
+high_water(Task_Type) ->
+  high_water(Task_Type, false).
+
+high_water(Task_Type, Reset) ->
+  try
+    if Reset -> - hd(ets:update_counter(?MODULE, Task_Type, [?HW_READ_CMD, ?HW_RESET_CMD]));
+       true  -> - ets:update_counter(?MODULE, Task_Type, ?HW_READ_CMD)
+    end
+  catch
+    error:badarg ->
+      {unknown_task_type, Task_Type}
+  end.
```

Comment:
This is the first introduction of try/catch to handle the case of a missing ets table. Proper handling of ets tables would require wrapping try/catch around *every single access* to *any* ets table for *reading or writing*. I've taken the strategy of crashing the caller with badarg. This may be a little difficult to figure out, but is consistent and doesn't require a bunch of defensive coding to obscure the algorithms. If try/catch is added in future, I want to do it as a complete consistent approach to the whole abstraction. The try/catch should be removed here.

Also, can you use ``case`` instead of ``if``? I would prefer the use of ``clear | no_clear`` or some other atomic options of your choice, rather than a boolean ``true | false``. This gives clearer semantic understanding of the API call, and it is not really a boolean() attribute, but a command.